app/Services/UserService.php

```diff
@@ -122,4 +126,39 @@ public function loginWithGoogle($data)
         return ['user' => $user, 'token' => $token];
     }
 
+    public function requestPasswordReset($email)
+    {
+        $user = User::where('email', $email)->first();
+        if (!$user) {
+            return response()->json(['error' => 'User not found'], 404);
+        }
+
+        $token = Str::random(64);
+        ForgetPasswordToken::create([
+            'email' => $email,
+            'token' => $token,
+        ]);
+
+        Mail::to($email)->send(new PasswordResetMail($token));
+
+        return response()->json(['message' => 'Password reset email sent']);
+    }
+
+    public function resetPassword($token, $newPassword)
+    {
+        $record = ForgetPasswordToken::where('token', $token)->first();
```

Comment:
here you should get the user and the token and make the query with finding user_id and token